Sources/TuistSupport/Atomic.swift

```diff
@@ -3,7 +3,7 @@ import Foundation
 /// Ensures that writing and reading from property annotated with this property wrapper is thread safe
 /// Taken from https://www.onswiftwings.com/posts/atomic-property-wrapper/
 @propertyWrapper
-struct Atomic<Value> {
+class Atomic<Value> {
```

Comment:
Stress testing some more I encountered the following crash :/

```
Generating workspace App.xcworkspace
Generating project App
Generating project C
Generating project A
Generating project D
Generating project B
tuist(1039,0x7000078d0000) malloc: Incorrect checksum for freed object 0x7fce6be077c8: probably modified after being freed.
Corrupt value: 0x4ffffffe00000000
tuist(1039,0x7000078d0000) malloc: *** set a breakpoint in malloc_error_break to debug
```

The only bit of shared mutable state I found that could cause this was `XcodeController` (used by `DefaultSettingsProvider`). It uses the `Atomic` property wrapper to synchronize access to a cached result.

Looking closer at `Atomic` - should it not be a reference type? (to ensure the underlying `nslock` isn't copied) 

On a hunch I made this change and repeated the stress test and it seems to have addressed that issue.

I used a local `generate-all.sh` script to generate all the fixtures in a loop (from within the `fixtures` directory):

```sh
#!/bin/bash -e

declare -a skipped_fixtures=(
    "invalid_workspace_manifest_name" 
    "ios_app_with_incompatible_dependencies" 
    "ios_app_with_incompatible_xcode" 
    "ios_workspace_with_dependency_cycle"
)

for i in {1..20}; do
    for f in *
    do
        if [[ ! " ${skipped_fixtures[@]} " =~ " ${f} " ]]; then
            if [[ -d $f ]]; then
                echo ">> Generating: $f"
                swift run tuist generate --path $f
                echo ""
            fi
        fi
    done
done
```